brick/__main__.py

```diff
@@ -382,23 +385,10 @@ def build(ctx, target, skip_previous_steps=None):
             else:
                 os.remove(host_path)
 
-        # Pull out the outputs to the host
-        container = docker_client.containers.run(image=digest, remove=True, detach=True)
-        try:
-            archive_bits, _stats = container.get_archive(container_path)
-        except docker.errors.NotFound:
-            # We see periodic failures, not sure if we should restart the container?
-            logger.info(f"Collecting failed, retrying")
-            time.sleep(2)
-            archive_bits, _stats = container.get_archive(container_path)
-
-        with io.BytesIO() as archive_tar_content:
-            for chunk in archive_bits:
-                archive_tar_content.write(chunk)
-            archive_tar_content.seek(0)
-            with tarfile.open(fileobj=archive_tar_content, mode="r:") as tar:
-                host_output_folder = os.path.abspath(os.path.join(host_path, "../"))
-                tar.extractall(host_output_folder)
+        host_output_folder = os.path.abspath(os.path.join(host_path, "../"))
+        container_id = run_shell_command(f"docker create {digest}")
```

Comment:
any risk of having stale containers if for some reason some commands fail here?
I don't think it's a problem, but still wanted to raise.
It's not blocking.

Reply:
As these containers are just writeable container layer over the specified image, then the only risk we have is wasting some disk space in case of output extraction fails.